### DISCUSSION TITLE
How to disable character animation when hitting notes?

### DISCUSSION THREAD

**Discussion by MAZ12211** [FunkinCrew/Funkin#3963] [Q&A]
I'm trying to make a notetype that prevents characters from playing their note direction animation

In Codename Engine, it had its own function called `cancelAnim()` for the strumline character

I don't know how to do it in base game though due to how little resources it currently has on notetypes

**Comment by KoloInDaCrib** [comment]
[The script for the character `bf-christmas`](https://github.com/FunkinCrew/funkin.assets/blob/main/preload/scripts/characters/bf-christmas.hxc) has a function onNoteHit which handles the logic for the censor animation. It looks like this: 
![image](https://github.com/user-attachments/assets/a5d74c07-0494-4cfb-b195-2b2fed60a7fc)

Basically what it does is it checks if a note with the type "censor" has been hit from the player side (while `bf-christmas` is used as the player character) and then it cancels any further behaviour within that character (which is to play the correct animation) and instead plays the custom animation with the suffix "censor". This is due to the fact that `super.onNoteHit(event);` is never called when the conditions for the censor animation are met, which is the line handling the logic behind animations.

To fully prevent the animation from being played, you can write something similar with the same principle, just without calling the function `playSingAnimation`

**Reply by THEkatinamicrowave** [reply]
Not "without calling" the function; probably more "cancelling" the function. playSingAnimation itself is hardcoded, so you'd have to cancel it.

**Comment by Lasercar** [comment]
https://github.com/FunkinCrew/Funkin/pull/4285
